### PR TITLE
Fix few memory leaks

### PR DESCRIPTION
**Pull request by DavBfr**
Fix some memory leaks observed in channels
* Printer Redirection
* Rdpsnd
* Rdpdr
* Disp

### CODE REVIEW

**Comment by hardening** [comment]
@freerdp-bot can you test this please ?

File: channels/rdpsnd/client/mac/rdpsnd_mac.c [hunk 1]
```diff
@@ -327,6 +327,7 @@ static void rdpsnd_mac_waveplay(rdpsndDevicePlugin* device, RDPSND_WAVE* wave)
 	length = wave->length > audioBuffer->mAudioDataBytesCapacity ? audioBuffer->mAudioDataBytesCapacity : wave->length;
     
 	CopyMemory(audioBuffer->mAudioData, wave->data, length);
+	free(wave->data);
 	audioBuffer->mAudioDataByteSize = length;
```
Comment (akallabeth) [inline channels/rdpsnd/client/mac/rdpsnd_mac.c]: Maybe a good idea to set the pointer NULL to force segfaults on invalid access

**Comment by DavBfr** [comment]
Set the pointer NULL.

**Comment by akallabeth** [comment]
@freerdp-bot test again please

**Comment by akallabeth** [comment]
@freerdp-bot be so kind and listen to me! test this stuff.

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by hardening** [comment]
@freerdp-bot come on lazy bot, test that !

**Pull request merged**